java-frontend/src/main/java/org/sonar/java/ast/parser/ListTreeImpl.java

```diff
@@ -67,45 +64,8 @@ public Kind kind() {
   }
 
   @Override
-  public Iterable<Tree> children() {
-    return new InterleaveIterable(list, separators);
-  }
-
-  private class InterleaveIterable implements Iterable<Tree> {
-
-    private final List<Iterator<? extends Tree>> iterators;
-
-    public InterleaveIterable(List<T> list, List<SyntaxToken> separators) {
-      iterators = Arrays.asList(list.iterator(), separators.iterator());
-    }
-
-    @Override
-    public Iterator<Tree> iterator() {
-      return new InterleaveIterator<>(iterators);
-    }
-  }
-
-  private static class InterleaveIterator<E> extends AbstractIterator<E> {
```

Comment:
so cool we can drop the AbstractIterator!

Reply:
I guess this class (fully covered btw :) ) lived for a couple of hours